chain/index/ddls.go

```diff
@@ -0,0 +1,98 @@
+package index
+
+import "database/sql"
+
+// TODO PRAGMA foreign_keys = ON;
+// TODO: Allow for msg_cid <> eth_tx_hash for unconfirmed mpool messages
+var ddls = []string{
+	`CREATE TABLE IF NOT EXISTS tipset_message (
+		message_id INTEGER PRIMARY KEY,
+		tipset_key_cid BLOB NOT NULL,
+		height INTEGER NOT NULL,
+		reverted INTEGER NOT NULL,
+		message_cid BLOB NOT NULL,
+		message_index INTEGER NOT NULL,
+		events_processed INTEGER NOT NULL,
+		UNIQUE (tipset_key_cid, message_cid)
+	)`,
+
+	`CREATE TABLE IF NOT EXISTS event (
+		event_id INTEGER PRIMARY KEY,
+		message_id INTEGER NOT NULL,
+        event_index INTEGER NOT NULL,
+        emitter_addr BLOB NOT NULL,
+		reverted INTEGER NOT NULL,
+		FOREIGN KEY (message_id) REFERENCES tipset_message(message_id) ON DELETE CASCADE,
+		UNIQUE (message_id, event_index)
+    )`,
+
+	`CREATE TABLE IF NOT EXISTS event_entry (
+		event_id INTEGER NOT NULL,
+		indexed INTEGER NOT NULL,
+		flags BLOB NOT NULL,
+		key TEXT NOT NULL,
+		codec INTEGER,
+		value BLOB NOT NULL,
+		FOREIGN KEY (event_id) REFERENCES event(event_id) ON DELETE CASCADE
+	)`,
+
+	`CREATE TABLE eth_tx_hash (
+		eth_tx_hash TEXT PRIMARY KEY,
```

Comment:
would a fixed-length `CHAR` buy us some efficiency here? I think we should be able to know with certainty what the length of every txhash is, right?

Reply:
ah, probably not, I think sqlite might fake its fixed-length fields